### PR TITLE
Get and download songs on Newgrounds

### PR DESCRIPTION
**Pull request by Syudagye**
Now you can download metadata of songs and songs with this :D

### CODE REVIEW

File: GDAPI/GDAPI/Application/Newgrounds/SongDownloader.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿using GDAPI.Objects.GeometryDash.General;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace GDAPI.Application.Newgrounds
+{
+    class SongDownloader
```
Comment (Altenhh) [inline GDAPI/GDAPI/Application/Newgrounds/SongDownloader.cs]: classes should always be public

File: GDAPI/GDAPI/Application/Newgrounds/SongDownloader.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        private async Task DownloadSong(SongMetadata song)
+        {
+            if (song == SongMetadata.Unknown)
+                return;
+            else
+            {
+                var filepath = $"{Database.GDLocalData}{Path.DirectorySeparatorChar}{song.ID}.mp3";
+                var res = await client.GetAsync(song.DownloadLink);
+                var bytes = await res.Content.ReadAsByteArrayAsync();
```
Comment (Altenhh) [inline GDAPI/GDAPI/Application/Newgrounds/SongDownloader.cs]:
```suggestion
                var bytes = await res.Content.ReadAsByteArrayAsync();
                
```

File: GDAPI/GDAPI/Functions/General/UsefulFunctions.cs [hunk 3]
```diff
@@ -38,5 +38,39 @@ public static void WriteAllLinesWithoutUselessNewLine(string path, List<StringBu
...
+        {
+            var chars = str.ToCharArray();
+            var result = new StringBuilder();
+            for (int i = 0; i < chars.Length; i++)
+            {
+                if (chars[i] == '%')
+                {
+                    switch (chars[i + 1].ToString() + chars[i + 2].ToString()) // This website for hex codes => https://www.obkb.com/dcljr/charstxt.html
+                    {
+                        case "3A": // colon (not the fox lol)
```
Comment (Altenhh) [inline GDAPI/GDAPI/Functions/General/UsefulFunctions.cs]:
```suggestion
                        case "3A": // colon
```

File: GDAPI/GDAPI/Program.cs [hunk 4]
```diff
@@ -1,10 +1,4 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace GDAPI
+﻿namespace GDAPI
```
Comment (Altenhh) [inline GDAPI/GDAPI/Program.cs]: there's no reason for this change here. create a secondary pull request to clean up some of the code if you want.

Comment (Syudagye) [reply GDAPI/GDAPI/Program.cs]: i just cleared unused dependencies, was that unnecessary ?

**Comment by Syudagye** [comment]
brw i don't really know how i can add a test for [this](https://github.com/Syudagye/GDAPI/blob/master/GDAPI/GDAPI/Application/Newgrounds/SongDownloader.cs)

**Pull request merged**